### PR TITLE
feat(radio-button): initial design guidance

### PR DESCRIPTION
**Pull request by venkateshr06**
Added initial guidance for Radio Component
[CDE-2361](https://vmw-jira.broadcom.net/browse/CDE-2361)-Create MD file for Radio component.

### CODE REVIEW

File: 1021-radio-design-guidance.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Guidance
+
+- **Must** use when only one item can be selected from a list.
+- **Must** NOT leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
+- **Must** use when the list includes at least two mutually exclusive options.
+- **Must** NOT use when more than 6 options are available to choose from. Consider using a `select` component instead.
+- **Must** NOT place interactive content (buttons, links, etc) inside the label.
+- **Should** NOT use for functionality like “accepting terms of service” or similar. Use a `checkbox` instead.
+- Use appropriate [states](https://clarity.design/documentation/radio#states) for Radio button.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:20]:
```suggestion
- **Should** use appropriate [radio button states](https://clarity.design/documentation/radio#states).
```

File: 1021-radio-design-guidance.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- Radio component is available in multiple [layouts](https://clarity.design/documentation/radio#layouts) variants.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:21]:
```suggestion
```

File: 1021-radio-design-guidance.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+[Radio](https://clarity.design/documentation/radio) component is a form element that allow the user to select one of the options in the list.
+
+## Guidance
+
+- **Must** use when only one item can be selected from a list.
+- **Must** NOT leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
+- **Must** use when the list includes at least two mutually exclusive options.
+- **Must** NOT use when more than 6 options are available to choose from. Consider using a `select` component instead.
+- **Must** NOT place interactive content (buttons, links, etc) inside the label.
+- **Should** NOT use for functionality like “accepting terms of service” or similar. Use a `checkbox` instead.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:19]:
```suggestion
- **Must not** use for functionality like “accepting terms of service” or similar. Use a [checkbox](https://clarity.design/documentation/checkbox) instead.
```

I think this should be a "must not". Also, added a link.

File: 1021-radio-design-guidance.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+[Radio](https://clarity.design/documentation/radio) component is a form element that allow the user to select one of the options in the list.
+
+## Guidance
+
+- **Must** use when only one item can be selected from a list.
+- **Must** NOT leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
+- **Must** use when the list includes at least two mutually exclusive options.
+- **Must** NOT use when more than 6 options are available to choose from. Consider using a `select` component instead.
+- **Must** NOT place interactive content (buttons, links, etc) inside the label.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:18]: This applies to all form control labels, not just radio buttons. Perhaps this should just be in the forms page.

File: 1021-radio-design-guidance.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+# Radio Component
+
+[Radio](https://clarity.design/documentation/radio) component is a form element that allow the user to select one of the options in the list.
+
+## Guidance
+
+- **Must** use when only one item can be selected from a list.
+- **Must** NOT leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
+- **Must** use when the list includes at least two mutually exclusive options.
+- **Must** NOT use when more than 6 options are available to choose from. Consider using a `select` component instead.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:17]:
```suggestion
- **Must not** use when more than 6 options are available to choose from. Consider using a [select](https://clarity.design/documentation/select) component instead.
```

File: 1021-radio-design-guidance.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+# Radio Component
+
+[Radio](https://clarity.design/documentation/radio) component is a form element that allow the user to select one of the options in the list.
+
+## Guidance
+
+- **Must** use when only one item can be selected from a list.
+- **Must** NOT leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:15]:
```suggestion
- **Must not** leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
```

File: 1021-radio-design-guidance.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-16
+updated: 2024-10-22
+---
+
+# Radio Component
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:8]:
```suggestion
# Radio Design Guidance
```

File: 1021-radio-design-guidance.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+[Radio](https://clarity.design/documentation/radio) component is a form element that allow the user to select one of the options in the list.
```
Comment (kevinbuhmann) [inline 1021-radio-design-guidance.md:10]:
```suggestion
A [radio](https://clarity.design/documentation/radio) control is a form element that allow the user to select one of the options in the list.
```

File: 1021-radio-design-guidance.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-16
+updated: 2024-10-22
+---
+
+# Radio Design Guidance
+
+A [radio](https://clarity.design/documentation/radio) control is a form element that allow the user to select one of the options in a group.
```
Comment (jbusta5) [inline 1021-radio-design-guidance.md:10]:
```suggestion
A [radio](https://clarity.design/documentation/radio) control is a form element that allows users to select one option from a group, ensuring clear choice among mutually exclusive options.```

File: 1021-radio-design-guidance.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+## Guidance
+
+- **Must** use when only one item can be selected from a list.
```
Comment (jbusta5) [inline 1021-radio-design-guidance.md:14]:
```suggestion
- Design teams **must** use radio buttons when only one item can be selected from a list.
```

File: 1021-radio-design-guidance.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+- **Must not** leave all radio buttons unselcted. Mark the first item or most preferred item as selected.
```
Comment (jbusta5) [inline 1021-radio-design-guidance.md:15]:
```suggestion
- *Design teams **must not** leave all radio buttons unselected. They should mark the first item or the most preferred item as selected by default.
```

File: 1021-radio-design-guidance.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+- **Must** use when the list includes at least two mutually exclusive options.
```
Comment (jbusta5) [inline 1021-radio-design-guidance.md:16]:
```suggestion
- Design teams **must** use radio buttons when the list includes at least two mutually exclusive options.
```

File: 1021-radio-design-guidance.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+- **Must not** use when more than 6 options are available to choose from. Consider using a [select](https://clarity.design/documentation/select) component instead.
```
Comment (jbusta5) [inline 1021-radio-design-guidance.md:17]:
```suggestion
- Design teams **must not** use radio buttons when more than six options are available. They should consider using a [select](https://clarity.design/documentation/select) component instead.
```

File: 1021-radio-design-guidance.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+- **Must not** use for functionality like “accepting terms of service” or similar. Use a [checkbox](https://clarity.design/documentation/checkbox) instead.  
+- **Should** use appropriate [radio button states](https://clarity.design/documentation/radio#states).
```
Comment (jbusta5) [inline 1021-radio-design-guidance.md:19]:
```suggestion
- Design teams **should** use appropriate [radio button states](https://clarity.design/documentation/radio#states) to provide clear feedback on user interactions and enhance the overall user experience.
```

**Pull request merged**